Classes/Driver/Version5/Mapping/NodeTypeMappingBuilder.php

```diff
@@ -0,0 +1,144 @@
+<?php
+namespace Flowpack\ElasticSearch\ContentRepositoryAdaptor\Driver\Version5\Mapping;
+
+/*
+ * This file is part of the Flowpack.ElasticSearch.ContentRepositoryAdaptor package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Flowpack\ElasticSearch\ContentRepositoryAdaptor\Driver\Version2;
+use Flowpack\ElasticSearch\Domain\Model\Index;
+use Flowpack\ElasticSearch\Domain\Model\Mapping;
+use Flowpack\ElasticSearch\Mapping\MappingCollection;
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Error\Result;
+use TYPO3\Flow\Error\Warning;
+use TYPO3\Flow\Object\ObjectManagerInterface;
+use TYPO3\TYPO3CR\Domain\Model\NodeType;
+
+/**
+ * NodeTypeMappingBuilder for Elasticsearch version 2.x
+ *
+ * @Flow\Scope("singleton")
+ */
+class NodeTypeMappingBuilder extends Version2\Mapping\NodeTypeMappingBuilder
+{
+    /**
+     * Called by the Flow object framework after creating the object and resolving all dependencies.
+     *
+     * @param integer $cause Creation cause
+     */
+    public function initializeObject($cause)
+    {
+        parent::initializeObject($cause);
+        if ($cause === ObjectManagerInterface::INITIALIZATIONCAUSE_CREATED) {
+            $this->adjustStringTypeMapping($this->defaultConfigurationPerType);
+        }
+    }
+
+    /**
+     * Builds a Mapping Collection from the configured node types
+     *
+     * @param Index $index
+     * @return MappingCollection<\Flowpack\ElasticSearch\Domain\Model\Mapping>
+     */
+    public function buildMappingInformation(Index $index)
+    {
+        $this->lastMappingErrors = new Result();
+
+        $mappings = new MappingCollection(MappingCollection::TYPE_ENTITY);
+
+        /** @var NodeType $nodeType */
+        foreach ($this->nodeTypeManager->getNodeTypes() as $nodeTypeName => $nodeType) {
+            if ($nodeTypeName === 'unstructured' || $nodeType->isAbstract()) {
+                continue;
+            }
+
+            $type = $index->findType(self::convertNodeTypeNameToMappingName($nodeTypeName));
+            $mapping = new Mapping($type);
+            $fullConfiguration = $nodeType->getFullConfiguration();
+            if (isset($fullConfiguration['search']['elasticSearchMapping'])) {
+                $fullMapping = $fullConfiguration['search']['elasticSearchMapping'];
+                $this->adjustStringTypeMapping($fullMapping);
+                $mapping->setFullMapping($fullMapping);
+            }
+
+            // https://www.elastic.co/guide/en/elasticsearch/reference/5.4/dynamic-templates.html
+            $mapping->addDynamicTemplate('dimensions', [
+                'path_match' => '__dimensionCombinations.*',
+                'match_mapping_type' => 'string',
+                'mapping' => [
+                    'type' => 'text'
```

Comment:
This should be `keyword`, since the value is always used for exact matches, it does not need to be analyzed. This leads to a mapping conflict, though (hence this "bugfix" of setting it to `text`), that needs to be solved properly.